libcrux-ml-dsa/src/arithmetic.rs

```diff
@@ -10,17 +10,60 @@ impl PolynomialRingElement {
         // FIXME: hax issue, 256 is COEFFICIENTS_IN_RING_ELEMENT
         coefficients: [0i32; 256],
     };
+
+    pub(crate) fn add(&self, rhs: &Self) -> Self {
+        let mut sum = Self::ZERO;
+
+        for i in 0..rhs.coefficients.len() {
+            sum.coefficients[i] = self.coefficients[i] + rhs.coefficients[i];
+        }
+
+        sum
+    }
+
+    pub(crate) fn sub(&self, rhs: &Self) -> Self {
+        let mut difference = Self::ZERO;
+
+        for i in 0..rhs.coefficients.len() {
+            difference.coefficients[i] = self.coefficients[i] - rhs.coefficients[i];
+        }
+
+        difference
+    }
+
+    pub(crate) fn infinity_norm_exceeds(&self, value: i32) -> bool {
+        if value > (FIELD_MODULUS - 1) / 8 {
+            return true;
+        }
+
+        // It is ok to leak which coefficient violates the bound since
+        // the probability for each coefficient is independent of secret
+        // data but we must not leak the sign of the centralized representative.
+        for coefficient in self.coefficients.iter() {
+            // Normalize the coefficient
+            let sign = coefficient >> 31;
+            let normalized = coefficient - (sign & (2 * coefficient));
+
+            if normalized >= value {
+                return true;
```

Comment:
I don't think this will go through hax.

Reply:
For both this and the other loop, I used a naive approach with a comment indicating we could revisit if performance is impacted.